### ISSUE TITLE
Customization on error response

### ISSUE THREAD

**Issue by ReaganM02** [Baroshem/nuxt-security#74]
How to properly handle response? I created an `~error.vue ` and it works for other error responses like 404 response but not on 429. 

### Version
nuxt-security: ^0.9.0
nuxt: 3.0.0

### What is Expected?
I was expecting to customize the error response like redirecting to another page or customize the default response.

### What is actually happening?
I am getting the default error response from nuxt
<img width="1913" alt="Screenshot 2022-12-16 at 2 45 58 AM" src="https://user-images.githubusercontent.com/21333078/208124792-e150cd69-137e-4f77-94b0-287c9c5ef82e.png">

Thank you

**Comment by Baroshem** [comment]
Hey, for that, I think we could implement another option in the config called handler. When passed it could replace the default functionality so that this method will handle the error. You will catch the error but dont throw so that you can do something else with it. What do you think about it?

**Comment by Baroshem** [comment]
Something like this:

```

security: {
    requestSizeLimiter: {
      value: {
        maxRequestSizeInBytes: 3000000,
        maxUploadFileRequestInBytes: 9000000,
      },
      route: '/upload-file',
      handler: () => console.error("Error")
    }
  }

```

**Comment by ReaganM02** [comment]
Thank you!  I tried but I am getting an error when adding the `handler` option
<img width="413" alt="Screenshot 2022-12-16 at 11 43 23 PM" src="https://user-images.githubusercontent.com/21333078/208135781-36333114-5f98-4350-9e37-6818f4eaf91a.png">

**Comment by ReaganM02** [comment]
My bad! I misunderstood your first response.😅 I'll wait for this. Thank you!

**Comment by Baroshem** [comment]
I cannot pass the method as it is removed from the config in the middleware. Instead, I will add an optional property called `throwError`. By default it has a value of true and will throw an error once certain condition is met (for example 429 after too many requests from rate limiter). But when set to false, it will return an object with the error but will not throw an error. Thanks to this approach you will be able to take this object and do whatever you want with it. For example show a custom error. I will create a pull request with this functionality for you to check out.

**Comment by Baroshem** [comment]
https://github.com/Baroshem/nuxt-security/pull/75

**Comment by Baroshem** [comment]
@ReaganM02 

Have you seen the PR that I have created? :)

**Comment by ReaganM02** [comment]
Hi @Baroshem , apologies for the delay response. I just got back from vacation. Is this already implemented?(sorry, I am still learning how github works. I'm still new to this.) I updated the version to `0.9.0` and I am getting this error inside the `nuxt.config.js` (see screenshot)
<img width="894" alt="Screenshot 2022-12-20 at 9 47 57 PM" src="https://user-images.githubusercontent.com/21333078/208682419-16921564-6a9e-4163-8c76-c5e105ef0b48.png">

`"nuxt-security": "^0.9.0",`

Thank you for this changes!! 🙏🏼

**Comment by Baroshem** [comment]
Hey @ReaganM02 

No, it is not implemented yet. You can test it out by cloning the repository and trying the playground Nuxt project with that change. I am looking forward to your feedback as it is crucial for me to make a decision whether to merge this PR or not 😉

**Comment by Baroshem** [comment]
@ReaganM02 

You are in the right place 😉

Are you using the correct branch?

**Comment by ReaganM02** [comment]
My bad! Yes I was not on the correct branch(Sorry). I tested it again and it looks like it is working. I am able to see the object response, the `statusCode` and the `statusMessage`. 

Server
<img width="528" alt="Screenshot 2022-12-21 at 2 38 26 AM" src="https://user-images.githubusercontent.com/21333078/208741612-047b1723-0d50-4b70-afb0-48159705bed1.png">

Client

<img width="416" alt="Screenshot 2022-12-21 at 2 39 21 AM" src="https://user-images.githubusercontent.com/21333078/208741705-079107cc-c9df-46a7-9a66-ce1999fa8e1b.png">

This is a very good idea for customization. I just need to find a way to catch this kind of response. So far, I tested different kinds of approaches and none of them works 😅

This is a very good update. Cheers!!

**Comment by Baroshem** [comment]
Have you tried this? https://nuxt.com/docs/getting-started/error-handling#rendering-an-error-page

**Comment by ReaganM02** [comment]
Here are the steps I tried. 

1. I created an `error.vue` file in the source directory alongside `app.vue file`
2. Inside `error.vue` 
```
<template>
	<div>
		{{ error }}
	</div>
</template>
  
<script setup>
const props = defineProps( {
	error: Object
} )

</script>
```

I tested it and the result will return the response JSON(The same as before)
  
<img width="402" alt="Screenshot 2022-12-21 at 10 22 29 PM" src="https://user-images.githubusercontent.com/21333078/208927362-00c3f206-f4e6-477d-ab75-f4bc087d6a21.png">

Because I added the `error.vue` file, when visiting to URL that don't exist it will show, this.  The `error.vue` is working
<img width="959" alt="Screenshot 2022-12-21 at 10 23 13 PM" src="https://user-images.githubusercontent.com/21333078/208927782-e929d3a2-fcd0-4b1c-b3f8-a203b53099f5.png">
